pkg/unidling/controller/unidling_controller.go

```diff
@@ -99,10 +100,27 @@ func NewUnidlingController(scaleNS scale.ScalesGetter, mapper meta.RESTMapper, e
 
 	_, controller := cache.NewInformer(
 		&cache.ListWatch{
-			// No need to list -- we only care about new events
+			// https://bugzilla.redhat.com/show_bug.cgi?id=1887745 -- making sure the ListFunc is
+			// using chunking for bootstrap and resync periods, meaning resourceVersion=0 is replaced
+			// by empty string, in order to have limits respected. Consider:
+			//    https://github.com/kubernetes/apiserver/blob/0e46f0ea2bdc958827712205cf9a88140a28409d/pkg/storage/cacher/cacher.go#L655-L668
 			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				fn := func(ctx context.Context, opts metav1.ListOptions) (runtime.Object, error) {
+					return evtNS.Events(metav1.NamespaceAll).List(ctx, opts)
+				}
+				p := pager.New(pager.ListPageFunc(fn))
+
 				options.FieldSelector = fieldSelector.String()
-				return evtNS.Events(metav1.NamespaceAll).List(context.TODO(), options)
+				options.Limit = 500
+
+				// skipping rv=0 to guarantee requests will take advantage of chunking
```

Comment:
would mention that `=0` means to list from cache and the cache does not support chunking.

Reply:
Indeed. Just amended the comment to mention zero means reading from cache. Thanks!